websites/views.py

```diff
@@ -167,6 +171,38 @@ def get_serializer_class(self):
         else:
             return WebsiteStarterDetailSerializer
 
+    @action(detail=False, methods=["post"], permission_classes=[])
+    def site_configs(self, request):
+        """Process webhook requests for WebsiteStarter site configs"""
+        data = json.loads(request.body)
+        if data.get("repository"):
+            if not valid_key(settings.GITHUB_WEBHOOK_KEY, request):
+                return Response(status=status.HTTP_403_FORBIDDEN)
+            files = [
+                file
+                for sublist in [
+                    commit["modified"] + commit["added"] for commit in data["commits"]
+                ]
+                for file in sublist
+                if file.endswith(settings.OCW_STUDIO_SITE_CONFIG_FILE)
+            ]
+            try:
+                succeeded, failed = sync_starter_configs(
```

Comment:
This makes some HTTP requests to fetch content from github. Should this function be wrapped with a task?

Reply:
I can do that, though in that case the response status should probably be changed to always be 202 with an empty body since the processing will be done asynchronously.